plaso/parsers/ps_transcript.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""Parser for powershell transcript files."""
+
+import copy
+import os
+import re
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import interface
+from plaso.parsers import manager
+
+
+class PSTranscriptEventData(events.EventData):
+  """ PowerShell transcript event data.
+
+  Attributes:
+    build_version (str): Build number of current version.
+    clr_version (str): Common Language Runtime version.
+    command (str): Command that was executed.
+    compatible_versions (str): Compatible PowerShell versions.
+    config_name (str): Configuration name.
+    edition (str): PowerShell edition
+    host_application (str): Application that executed command.
+    machine (str): Hostname of machine.
+    process_id (str): Process ID.
+    remoting_protocol_version (str): PowerShell
+      remote management protocol version.
+    runas_user (str): User context of execution.
+    serialization_version (str): Serialization method version.
+    username (str): User that executed command.
+    version (str): PowerShell version.
+    ws_man_stack_version (str): WS-Management stack version
+    username (str): User that executed command.
+  """
+
+  DATA_TYPE = 'ps:transcript:event'
+
+  def __init__(self):
+    """Initializes event data."""
+
+    super(PSTranscriptEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.build_version = None
+    self.clr_version = None
+    self.command = None
+    self.compatible_versions = None
+    self.config_name = None
+    self.edition = None
+    self.host_application = None
+    self.machine = None
+    self.process_id = None
+    self.remoting_protocol_version = None
+    self.runas_user = None
+    self.serialization_version = None
+    self.username = None
+    self.version = None
+    self.ws_man_stack_version = None
+
+
+class PSTranscriptParser(interface.FileObjectParser):
```

Comment:
One question about the choice of interface here. Is there a particular reason you are not using one of the text_parser classes (e.g. PyparsingMultiLineTextParser) and a pyparsing grammar? Transcript files seem to be UTF-8 encoded text files. It might make it easier to parse and verify the multi line transcript events.

Reply:
I think the example in the documentation uses interface.FileObjectParser because the example deals with a binary parser. In this case, i think a pyparsing parser makes more sense. @joachimmetz thoughts?